client.go

```diff
@@ -67,6 +67,14 @@ func (cl *Client) Do(c context.Context, req *Request) (*Response, error) {
 		if resp.response.StatusCode < 500 {
 			break
 		}
+
+		// close the response body before we lose our reference to it
+		if err = resp.response.Body.Close(); err != nil {
```

Comment:
I'm now closing the > 500 response body on each loop... I actually need to change this so that it doesn't close the body on the last iteration